src/plugins/workspace/public/components/service_card/workspace_list_card.tsx

```diff
@@ -0,0 +1,239 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { Component } from 'react';
+import {
+  EuiPanel,
+  EuiLink,
+  EuiDescriptionList,
+  EuiIcon,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiSelect,
+  EuiButtonIcon,
+  EuiSpacer,
+  EuiListGroup,
+  EuiText,
+  EuiTitle,
+  EuiToolTip,
+  EuiEmptyPrompt,
+} from '@elastic/eui';
+import { i18n } from '@osd/i18n';
+import { Subscription } from 'rxjs';
+import moment from 'moment';
+import _ from 'lodash';
+import { CoreStart, WorkspaceObject } from '../../../../../core/public';
+import { navigateToWorkspaceDetail } from '../utils/workspace';
+
+import { WORKSPACE_CREATE_APP_ID, WORKSPACE_LIST_APP_ID } from '../../../common/constants';
+import { WorkspaceEntry, recentWorkspaceManager } from '../../recent_workspace_manager';
+
+const WORKSPACE_LIST_CARD_DESCRIPTIOIN = i18n.translate('workspace.list.card.descriptionh', {
+  defaultMessage:
+    'Workspaces are dedicated environments for organizing and collaborating on your data, dashboards, and analytics workflows. Each Workspace acts as a self-contained space with its own set of saved objects and access controls.',
+});
+
+const MAX_ITEM_IN_LIST = 5;
+
+export interface WorkspaceListCardProps {
+  core: CoreStart;
+}
+
+export interface WorkspaceListItem {
+  id: string;
+  name: string;
+  time?: string | number;
+}
+
+export interface WorkspaceListCardState {
+  availiableWorkspaces: WorkspaceObject[];
+  filter: string;
+  workspaceList: WorkspaceListItem[];
+  recentWorkspaces: WorkspaceEntry[];
+}
+
+export class WorkspaceListCard extends Component<WorkspaceListCardProps, WorkspaceListCardState> {
+  private workspaceSub?: Subscription;
+  constructor(props: WorkspaceListCardProps) {
+    super(props);
+    this.state = {
+      availiableWorkspaces: [],
+      recentWorkspaces: [],
+      workspaceList: [],
+      filter: 'viewed',
+    };
+  }
+
+  componentDidMount() {
+    this.setState({
+      recentWorkspaces: recentWorkspaceManager.getRecentWorkspaces() || [],
+    });
+    this.workspaceSub = this.props.core.workspaces.workspaceList$.subscribe((list) => {
+      this.setState({
+        availiableWorkspaces: list || [],
+      });
+    });
+    this.loadWorkspaceListItems();
+  }
+
+  componentDidUpdate(
+    prevProps: Readonly<WorkspaceListCardProps>,
+    prevState: Readonly<WorkspaceListCardState>
+  ): void {
+    if (
+      !_.isEqual(prevState.filter, this.state.filter) ||
+      !_.isEqual(prevState.availiableWorkspaces, this.state.availiableWorkspaces) ||
+      !_.isEqual(prevState.recentWorkspaces, this.state.recentWorkspaces)
+    ) {
+      this.loadWorkspaceListItems();
+    }
+  }
+
+  private loadWorkspaceListItems() {
+    if (this.state.filter === 'viewed') {
+      this.setState({
+        workspaceList: _.orderBy(this.state.recentWorkspaces, ['timestamp'], ['desc'])
```

Comment:
It seems `workspaceList` can be derived from existing state, I think we can simply compute it in the `render()` function.

If you use functional react component, you can use `useMemo` to avoid unnecessary recompute. Btw, why not use functional component? I can see it will make it implementation much simple

Reply:
> Btw, why not use functional component? I can see it will make it implementation much simple

when i try using function component, dashboard embeddable not able to render it

Reply:
Oh, that's odd, then I think you can keep the current implementation, I will investigate a bit on this, thanks ;)

Reply:
thanks to @ruanyl help on debugging, seems functional component don't any issue, convert to functional component now.